### PR TITLE
Hereditary ontology repository

### PR DESCRIPTION
**Pull request by mntlra**
This repository represents the HEREDITARY Ontology.

The HEREDITARY Ontology is the backbone of the Semantic Data Integration of the HEREDITARY project (https://hereditary-project.eu/), which exploits the Ontology-Based Data Access (OBDA) technology to query, aggregate, and join large heterogeneous data in a distributed manner using a unique query language, i.e., SPARQL. The HEREDITARY Ontology has been designed to integrate ALS and MS data coming from different medical centers while avoiding sharing any raw data. The HEREDITARY Ontology models both clinical and genomics data.

### CODE REVIEW

File: hereditary/htaccess.txt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+# Rewrite rule to deal with two namespaces: schema and named-individual
+RewriteRule ^ontology/schema/(.*)$ http://hereditary.dei.unipd.it/ontology/#$1 [R=302,L,NE]
+RewriteRule ^ontology/resource/(.*)$ http://hereditary.dei.unipd.it/ontology/#https://w3id.org/hereditary/ontology/resource/$1 [R=302,L,NE]
+RewriteRule ^ontology/phenoclinical/schema/(.*)$ http://hereditary.dei.unipd.it/ontology/phenoclinical/#$1 [R=302,L,NE]
+RewriteRule ^ontology/phenoclinical/resource/(.*)$ http://hereditary.dei.unipd.it/ontology/#https://w3id.org/hereditary/ontology/phenoclinical/resource/$1 [R=302,L,NE]
+RewriteRule ^ontology/genomics/schema/(.*)$ http://hereditary.dei.unipd.it/ontology/genomics/#$1 [R=302,L,NE]
+RewriteRule ^ontology/genomics/resource/(.*)$ http://hereditary.dei.unipd.it/ontology/#https://w3id.org/hereditary/ontology/genomics/resource/$1 [R=302,L,NE]
+
+RewriteRule ^ontology(.*)$ http://hereditary.dei.unipd.it/ontology$1 [R=302,L]
+RewriteRule ^ontology/phenoclinical(.*)$ http://hereditary.dei.unipd.it/ontology/phenoclinical$1 [R=302,L]
```
Comment (dgarijo) [inline hereditary/htaccess.txt:50]: This rule is included in the previous one

File: hereditary/htaccess.txt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+RewriteCond %{HTTP_ACCEPT} application/xhtml\+xml [OR]
+RewriteCond %{HTTP_USER_AGENT} ^Mozilla/.*
+RewriteRule ^ontology$ http://hereditary.dei.unipd.it/ontology/
+RewriteRule ^ontology/phenoclinical$ http://hereditary.dei.unipd.it/ontology/phenoclinical/
+RewriteRule ^ontology/genomics$ http://hereditary.dei.unipd.it/ontology/genomics/
+
+# Rewrite rule to serve RDF/XML content from the vocabulary URI if requested
+RewriteCond %{HTTP_ACCEPT} application/rdf\+xml
+RewriteRule ^ontology/phenoclinical$ http://hereditary.dei.unipd.it/ontology/phenoclinical/schema/hero_phenoclinical.owl [R=303]
+RewriteRule ^ontology/genomics$ http://hereditary.dei.unipd.it/ontology/genomics/schema/hero_genomics.owl [R=303]
```
Comment (dgarijo) [inline hereditary/htaccess.txt:30]: the rewrite condition is per rule. You have to repeat the condition, otherwise this won't work as you intend

**Comment by dgarijo** [comment]
Please see my comments. I think it would be better to create subfolders and separate htaccess files for the corresponding ontos. Otherwise this file may be too big to maintain.

File: hereditary/htaccess.txt
Comment (davidlehn) [inline hereditary/htaccess.txt:1]: Rename this file to `.htaccess`.

File: hereditary/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# /HEREDITARY Ontology/
+This [W3ID](https://w3id.org) provides a persistent URI namespace for the HEREDITARY Ontology. 
+This project has received funding from the HEREDITARY Project, as part of the European Union’s Horizon Europe research and innovation programme under grant agreement No GA 101137074.
+
+## Contact:
+* **Laura Menotti** (<laura.menotti@unipd.it>)
+* **Gianmaria Silvello** (<silvello@dei.unipd.it>)
```
Comment (davidlehn) [inline hereditary/README.md:7]: Please add github user ids if available.  (We use those to check who is allowed to make changes.)

File: hereditary/htaccess.txt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+# Directive to ensure *.rdf files served as appropriate content type,
+# if not present in main apache config
+AddType application/rdf+xml .rdf
+AddType application/rdf+xml .owl
+AddType text/turtle .ttl
+
+# Rewrite engine setup
+RewriteEngine On
+RewriteBase /def
```
Comment (davidlehn) [inline hereditary/htaccess.txt:16]: This isn't needed.  It's only used when redirecting to local files.

File: hereditary/README.md (hunk 3 above)
Comment (TallTed) [reply hereditary/README.md:7]:
```suggestion
* **Laura Menotti** (<laura.menotti@unipd.it>) @mntlra
* **Gianmaria Silvello** (<silvello@dei.unipd.it>) @giansilv
```

**Comment by mntlra** [comment]
Dear all,

thank you for your feedback. I made the requested changes. 

Laura

**Comment by dgarijo** [comment]
@mntlra, I think you are not creating what you intend.
By having `^ontology/genomics/` you are minting  `https://w3id.org/hereditary/genomics/ontology/genomics/`. Is this what you want?
If you aim for `https://w3id.org/hereditary/genomics/ontology/` then remove the `/genomics` from your htaccess file. The same for the other subfolders.

**Comment by dgarijo** [comment]
Then you need to place the genomics folder under a `ontology` folder, and then modify the rules to be
`RewriteRule ^ontology/genomics/$ http://hereditary.dei.unipd.it/ontology/genomics/` --> `RewriteRule ^$ http://hereditary.dei.unipd.it/ontology/genomics/`

File: hereditary/genomics/.htaccess [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+AddType application/rdf+xml .owl
+AddType text/turtle .ttl
+
+# Rewrite rule to serve HTML content from the vocabulary URI if requested
+RewriteCond %{HTTP_ACCEPT} !application/rdf\+xml.*(text/html|application/xhtml\+xml)
+RewriteCond %{HTTP_ACCEPT} text/html [OR]
+RewriteCond %{HTTP_ACCEPT} application/xhtml\+xml [OR]
+RewriteCond %{HTTP_USER_AGENT} ^Mozilla/.*
+RewriteRule ^ontology/genomics/$ http://hereditary.dei.unipd.it/ontology/genomics/
+RewriteRule ^ontology/genomics(.*)$ http://hereditary.dei.unipd.it/ontology/genomics$1 [R=302,L]
```
Comment (dgarijo) [inline hereditary/genomics/.htaccess:20]: Also here, the rewrite rul won't apply to the rule in line 20. You have to add the condition again

**Comment by mntlra** [comment]
Thank you for the assistance. I made the requested changes.

**Comment by mntlra** [comment]
Dear all,

The redirect seems to not work as expected. From https://w3id.org/hereditary/ontology/phenoclinical/ I should be redirected to https://hereditary.dei.unipd.it/ontology/phenoclinical, instead the redirected URL is http://hereditary.dei.unipd.it/ontology/phenoclinical/http://hereditary.dei.unipd.it/ontology/phenoclinical/. How can I modify the .htaccess once it is merged?

**Comment by dgarijo** [comment]
you may modify it and start a new pull request

**Pull request merged**